src/main/java/org/sonar/ucfg/UCFGtoProtobuf.java

```diff
@@ -72,47 +72,80 @@ public static void toProtobufFile(UCFG ucfg, String filename) throws IOException
   }
 
   private static Ucfg.Instruction toProtobuf(UCFGElement.AssignCall assignCall) {
-    return Ucfg.Instruction.newBuilder().setAssigncall(Ucfg.AssignCall.newBuilder().setLocation(toProtobuf(assignCall.location()))
-      .setVariable(assignCall.getLhs().id())
+    Ucfg.AssignCall.Builder builder = Ucfg.AssignCall.newBuilder()
+      .setLocation(toProtobuf(assignCall.location()))
       .setMethodId(assignCall.getMethodId())
-      .addAllArgs(assignCall.getArgExpressions().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()))
-      .build()).build();
+      .addAllArgs(assignCall.getArgExpressions().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()));
+    Expression lhs = assignCall.getLhs();
+
+    if (lhs instanceof Expression.Variable) {
+      builder.setVariable(toProtobuf((Expression.Variable) lhs));
+    } else {
+      builder.setFieldAccess(toProtobuf((Expression.FieldAccess) lhs));
+    }
+    return Ucfg.Instruction.newBuilder().setAssigncall(builder.build()).build();
   }
 
   private static Ucfg.Instruction toProtobuf(UCFGElement.NewObject newObject) {
-    return Ucfg.Instruction.newBuilder().setNewObject(Ucfg.NewObject.newBuilder().setLocation(toProtobuf(newObject.location()))
-      .setVariable(newObject.getLhs().id())
-      .setType(newObject.instanceType())
-      .build()).build();
+    Ucfg.NewObject.Builder builder = Ucfg.NewObject.newBuilder()
+      .setLocation(toProtobuf(newObject.location()))
+      .setType(newObject.instanceType());
+
+    Expression lhs = newObject.getLhs();
+    if (lhs instanceof Expression.Variable) {
```

Comment:
lines 95-99 seem to be duplicating lines 81-85 - could move to a private method?

Reply:
Unfortunately not. The builders which are used as not the same type (`Ucfg.NewObject.Builder` vs `Ucfg.AssignCall.Builder`), and they are not implementing any common interface (generated code from UCFG). So even if the logic is exactly the same, I can not extract it to be reused... 

Which is sad, because the `setVariable` and `setFieldAccess` share the same signatures, but can not be replaced by a parent method. The only way would be to change the protobuf messages to had a LHS part, being a variable or a fieldAccess, but I'm not sure it will simplify things.